packages/datadog-plugin-mysql/src/index.js

```diff
@@ -23,14 +23,15 @@ class MySQLPlugin extends DatabasePlugin {
       }
     })
   }
-}
 
-function getServiceName (config, dbConfig) {
-  if (typeof config.service === 'function') {
-    return config.service(dbConfig)
-  }
+  getServiceName (dbConfig) {
+    if (typeof this.config.service === 'function') {
+      this.config.service = this.config.service(dbConfig)
```

Comment:
This will not work because if you have 3 instances then all 3 will end up with the same service name. The whole point of the option is to dynamically return a different service name based on the configuration options.

Reply:
Completely gutted the PR and removed all the caching stuff